code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -184,11 +186,21 @@ void setName(Map<String, Object> inspect, Instance instance) {
     }
 
     void setNetwork(Map<String, Object> inspect, Instance instance) {
-        String networkKind = KIND_NETWORK;
-        if (BooleanUtils.toBoolean(getRancherNetworkLabel(inspect))) {
-            networkKind = KIND_HOSTONLY;
+        Network network = null;
+        Network rancherNetwork = networkDao.getNetworkForObject(instance, KIND_HOSTONLY);
+        String ip = getDockerIp(inspect);
+        if (StringUtils.isNotEmpty(ip) && isIpInNetwork(ip, rancherNetwork)) {
+            DataAccessor.fields(instance).withKey(FIELD_REQUESTED_IP_ADDRESS).set(ip);
+            network = rancherNetwork;
+        }
+
+        if (network == null && BooleanUtils.toBoolean(getRancherNetworkLabel(inspect))) {
+            network = rancherNetwork;
+        }
+
+        if (network == null) {
+            network = networkDao.getNetworkForObject(instance, KIND_NETWORK);
```

Comment:
I know this logic isn't new, but I don't understand it.  If the user didn't ask for rancher network, what will this method assign it to?  The "unmanaged" network?